Automattic/wp-calypso
# Cleanup: remove segmented signup experiment

This cleans up the changes that were done to introduce segmented signup. 

Context here: p1723839885121359-slack-C072ZS50A81

## Proposed Changes

* This cleans up all code that was added to run the experiment and reverts the main onboarding flow to its original shape.

## Why are these changes being made?
The experiment didn't make a positive impact.

## Testing Instructions

Please test thoroughly.

1. Go through the onboarding flow by visiting /start.
2. Make a site with a free domain and a free plan.
3. Make a site with a free domain and a paid plan.
4. When picking a paid domain, the free plan should cease being a column, and should be in the subheader.
     - When clicking the CTA in the subheader, you should see a modal "A paid plan is required for your domain."
     - Please try all three options in the modal by going through the flow 3 times.
5. Pick a paid domain and a paid plan. You should land in checkout.
